benchmark/benchmark.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Benchmarking tool for PennyLane.
+"""
+# pylint: disable=import-outside-toplevel,invalid-name
+import argparse
+import importlib
+import subprocess
+import time
+
+import numpy as np
+
+import pennylane as qml
+
+
+# benchmarking tool version
+__version__ = "0.1.0"
+
+
+def timing(func, *, number=10, repeat=5):
+    """Time the given function.
+
+    Args:
+        func (Callable[[], Any]): function to time
+        number (int): number of loops per test
+        repeat (int): the number of times the timing test is run
+    """
+    import timeit
+
+    print('{} loops, {} runs'.format(number, repeat))
+    res = timeit.repeat(func, number=number, repeat=repeat, globals=globals())
+    print("Timing per loop:", np.array(res) / number)
```

Comment:
This is actually a great utility, I can never remember how to use `timeit` outside of IPython's magic `%timeit`!